src/ledger/NetworkConfig.cpp

```diff
@@ -868,6 +957,92 @@ SorobanNetworkConfig::ledgerMaxTxCount() const
     return mLedgerMaxTxCount;
 }
 
+uint32_t
+SorobanNetworkConfig::getBucketListSizeSnapshotPeriod() const
+{
+#ifdef BUILD_TESTS
+    if (mBucketListSnapshotPeriodForTesting)
+    {
+        return *mBucketListSnapshotPeriodForTesting;
+    }
+#endif
+
+    return BUCKETLIST_SIZE_SNAPSHOT_PERIOD;
+}
+
+void
+SorobanNetworkConfig::maybeUpdateBucketListWindowSize(AbstractLedgerTxn& ltx,
+                                                      uint32_t newSize) const
+{
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
+    // // Check if BucketList size window should exist
+    if (protocolVersionIsBefore(ltx.loadHeader().current().ledgerVersion,
+                                ProtocolVersion::V_20))
+    {
+        return;
+    }
+
+    auto currSize = mBucketListSizeSnapshots.size();
+    if (newSize == currSize)
+    {
+        // No size change, nothing to update
+        return;
+    }
+
+    if (newSize < currSize)
+    {
+        while (mBucketListSizeSnapshots.size() != newSize)
+        {
+            mBucketListSizeSnapshots.pop_front();
+        }
+    }
+    // If newSize > currSize, initialize new slots with current BucketList size
+    else
+    {
+        auto oldestSize = mBucketListSizeSnapshots.front();
+        while (mBucketListSizeSnapshots.size() != newSize)
+        {
+            mBucketListSizeSnapshots.push_back(oldestSize);
```

Comment:
Shouldn't this be `push_front`? (again, these should be the 'old' slots and you'd want to pop them ASAP, right?)